lib/issues.go

```diff
@@ -109,17 +97,18 @@ func DidIssueChange(config cfg.Config, ghIssue github.Issue, jIssue jira.Issue)
 // UpdateIssue compares each field of a GitHub issue to a JIRA issue; if any of them
 // differ, the differing fields of the JIRA issue are updated to match the GitHub
 // issue.
-func UpdateIssue(config cfg.Config, ghIssue github.Issue, jIssue jira.Issue, ghClient clients.GitHubClient, jClient clients.JIRAClient) error {
+func UpdateIssue(config cfg.Config, ghIssue models.ExtendedGithubIssue, jIssue models.ExtendedJiraIssue, ghClient issuesyncgithub.Client, jClient issuesyncjira.Client) error {
 	log := config.GetLogger()
 
 	log.Debugf("Updating JIRA %s with GitHub #%d", jIssue.Key, *ghIssue.Number)
 
 	var issue jira.Issue
 
-	if DidIssueChange(config, ghIssue, jIssue) {
+	if DidIssueChange(config, ghIssue, jIssue.Issue) {
 		fields := jira.IssueFields{}
 		fields.Unknowns = map[string]interface{}{}
 
+		fields.Status = jIssue.AvailableStatusesByName[ghIssue.ProjectCard.GetColumnName()]
```

Comment:
I'm pretty sure this won't work. Jira's api for dealing with Status is based around "transitions". I'm currently researching the API details and thinking about how I can deal with this api impedance mismatch.